### DISCUSSION TITLE
Apply same balance weights to different samples

### DISCUSSION THREAD

**Discussion by xinyangbing** [open2c/cooler#368] [Q&A] [answered]
Hi,
We have generated some Micro-C datasets where we are interested in changes in interactions between two loci with 200/400bp resolution bins. After ICE balancing, we found that this bin, which seems to have intrinsic low mappability (discovered by computing marginals), behaved erratically because of the lower coverage in one replicate vs the controls or the other rep of mutant. This seems to be arising possibly from subtle differences in MNase digestion for this replicate. It seems that the balancing is over-compensating for the under-sequencing of this bin in this replicate. Is there a way to extract the balance weight values used in one sample, and apply the same balance weight values to other samples in parallel in CLI? Thank you in advance.

**Comment by nvictus** [comment, marked as answer]
There is no CLI tool for this yet, but it's a just a few lines of Python to copy bin table columns between files, assuming the bin tables are aligned.

```python
import cooler

# optional
h5opts = dict(compression='gzip', compression_opts=6)

clr1 = cooler.Cooler(f'{path1}::resolutions/{binsize}')
weights = clr1.bins()['weight'][:].values

clr2 = cooler.Cooler(f'{path2}::resolutions/{binsize}')
with clr2.open('r+') as f:
    # Create a weight column
    f['bins'].create_dataset('weight', data=weights, **h5opts)
    
    # .. or overwrite an existing weight column
    f['bins/weight'][:] = weights
```
